lib/ClangImporter/ClangIncludePaths.cpp

```diff
@@ -296,14 +297,119 @@ getLibStdCxxFileMapping(ASTContext &ctx) {
+  if (llvm::getWindowsSDKDir(VFS, {}, {}, {},
+                             WindowsSDK.Path, WindowsSDK.MajorVersion,
+                             WindowsSDK.IncludeVersion,
+                             WindowsSDK.LibraryVersion)) {
+    llvm::SmallString<261> WinSDKInjection{WindowsSDK.Path};
+    llvm::sys::path::append(WinSDKInjection, "Include");
+    if (WindowsSDK.MajorVersion > 8)
+      llvm::sys::path::append(WinSDKInjection, WindowsSDK.IncludeVersion, "um");
+    llvm::sys::path::append(WinSDKInjection, "module.modulemap");
+
+    AuxiliaryFile = GetWindowsAuxiliaryFile("winsdk.modulemap", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(WinSDKInjection), AuxiliaryFile);
+  }
+
+  struct {
+    std::string Path;
+    std::string Version;
+  } UCRTSDK;
+  if (llvm::getUniversalCRTSdkDir(VFS, SearchPathOpts.getWinSDKRoot(),
+                                  SearchPathOpts.getWinSDKVersion(), {},
+                                  UCRTSDK.Path, UCRTSDK.Version)) {
+    llvm::SmallString<261> UCRTInjection{UCRTSDK.Path};
+    llvm::sys::path::append(UCRTInjection, "Include", UCRTSDK.Version, "ucrt");
+    llvm::sys::path::append(UCRTInjection, "module.modulemap");
+
+    AuxiliaryFile = GetWindowsAuxiliaryFile("ucrt.modulemap", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(UCRTInjection), AuxiliaryFile);
+  }
+
+  struct {
+    std::string Path;
+    llvm::ToolsetLayout Layout;
+  } VCTools;
+  if (llvm::findVCToolChainViaCommandLine(VFS, {}, {}, {}, VCTools.Path, VCTools.Layout) ||
+      llvm::findVCToolChainViaEnvironment(VFS, VCTools.Path, VCTools.Layout) ||
+      llvm::findVCToolChainViaSetupConfig(VFS, VCTools.Path, VCTools.Layout) ||
+      llvm::findVCToolChainViaRegistry(VCTools.Path, VCTools.Layout)) {
+    llvm::SmallString<261> VCToolsInjection{VCTools.Path};
+    llvm::sys::path::append(VCToolsInjection, "include");
+
+    llvm::sys::path::append(VCToolsInjection, "module.modulemap");
+    AuxiliaryFile =
+        GetWindowsAuxiliaryFile("vcruntime.modulemap", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(VCToolsInjection), AuxiliaryFile);
+
+    llvm::sys::path::remove_filename(VCToolsInjection);
+    llvm::sys::path::append(VCToolsInjection, "vcruntime.apinotes");
+    AuxiliaryFile =
+        GetWindowsAuxiliaryFile("vcruntime.apinotes", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(VCToolsInjection), AuxiliaryFile);
+  }
+
+  return Mappings;
+}
+}
+
 SmallVector<std::pair<std::string, std::string>, 2>
 swift::getClangInvocationFileMapping(ASTContext &ctx) {
   SmallVector<std::pair<std::string, std::string>, 2> result;
 
+  // Linux Mappings
   result.append(getGlibcFileMapping(ctx));
-
-  if (ctx.LangOpts.EnableCXXInterop) {
+  if (ctx.LangOpts.EnableCXXInterop)
     result.append(getLibStdCxxFileMapping(ctx));
-  }
+
+  // Windows Mappings
```

Comment:
Same for this line. The function name already explains itself.

Reply:
Done